src/app/reporting/ReportScheduler.h

```diff
@@ -167,14 +188,39 @@ class ReportScheduler : public ReadHandler::Observer
 
     virtual void ReportTimerCallback() = 0;
 
+    /// @brief Method that triggers a report emission on each ReadHandler that are above their min interval
+    ///        It checks if all read handlers are above their min interval and if so forces them in a dirty state to trigger a
+    ///        report emission
+    void TriggerReportEmission()
+    {
+        Timestamp now = mTimerDelegate->GetCurrentMonotonicTimestamp();
+        mNodesPool.ForEachActiveObject([now](ReadHandlerNode * node) {
+            if (now >= node->GetMinTimestamp())
+            {
+                node->GetReadHandler()->ForceDirtyState();
```

Comment:
But with this setup... this means we will send the empty reports on every wakeup, even if clients are OK with it taking a lot longer, right?  Why do we want to do that?

Reply:
It's to prevent the case where we wake up for a user action, go back to sleep, then wake up a short time after because the max timestamp was reached.

Here if we emit report now we are giving the device a chance to go to sleep for the max interval if nothing else happens.

Reply:
we might have to add conditions. The default idleInterval and activeInterval are really short. 
I believe this make sense for LIT but maybe not for SIT. Or at least for device that have a rather short idleTime, else we will be constantly sending reports.

Reply:
@lpbeliveau-silabs I think if we wake up, and all the max intervals are at least our sleep period out, we should not do any sending of anything...  because we know we will wake up again before any of that has to happen.

Reply:
The feature is really more for LIT ICDs were controllers buffer interaction until they receive either a check-in message or report.

I think having it behind a configuration flag for now might be the best option until we implement LIT ICDs and have a more cohesive solution.

Reply:
@bzbarsky-apple The problem is that for LIT ICD, we will want to send reports as soon as we enter active modes regardless of any other conditions.

For now it is ifdefed out until we further define the optimal reporting behavior since it is mostly important for LIT ICD.